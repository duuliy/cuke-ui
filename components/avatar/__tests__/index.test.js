import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Avatar from "../index";
import { UserIcon } from "../../icon";

describe("<Avatar/>", () => {
  it("should render a <Avatar/> components", () => {
    const wrapper = render(
      <div>
        <Avatar icon={<UserIcon />} />
        <Avatar text="黄瓜ui" />
        <Avatar src="https://www.test.jpg" />
        <Avatar icon={<UserIcon />} />
        <Avatar icon={<UserIcon />} shape="square" />
        <Avatar text="大" size="large" />
        <Avatar text="中" />
        <Avatar text="小" size="small" />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-avatar classnames", () => {
    const wrapper = render(
      <div>
        <Avatar icon={<UserIcon />} />
        <Avatar text="黄瓜ui" />
        <Avatar text="黄瓜ui" shape="square" />
        <Avatar src="https://gw.alipayobjects.com/zos/rmsportal/BiazfanxmamNRoxxVxka.png" />
        <Avatar text="黄瓜ui" size="small" />
        <Avatar text="黄瓜ui" size="large" />
      </div>
    );
    assert(wrapper.find(".cuke-avatar").length >= 1);
    assert(wrapper.find(".cuke-avatar-circle").length >= 1);
    assert(wrapper.find(".cuke-avatar-square").length === 1);
    assert(wrapper.find(".cuke-avatar-icon").length === 1);
    assert(wrapper.find(".cuke-avatar-text").length >= 1);
    assert(wrapper.find(".cuke-avatar-image").length === 1);
    assert(wrapper.find(".cuke-avatar-size-small").length === 1);
    assert(wrapper.find(".cuke-avatar-size-large").length === 1);
  });

  it("should render a icon", () => {
    const wrapper = shallow(<Avatar icon={<UserIcon />} />);
    assert(wrapper.find(".cuke-avatar-icon").length === 1);
    assert(wrapper.find(UserIcon).length === 1);
  });

  it("should render a image", () => {
    const src = "https://www.test.png";
    const wrapper = shallow(<Avatar src={src} alt="user" />);
    assert(wrapper.find(".cuke-avatar-image").length === 1);
    assert(wrapper.find("img").length === 1);
  });

  it("should render a text", () => {
    const wrapper = shallow(<Avatar text="user" />);
    assert(wrapper.find(".cuke-avatar-text").length === 1);
    assert(wrapper.find("span").length >= 1);
    expect(wrapper.text()).toEqual("user");
  });
});
