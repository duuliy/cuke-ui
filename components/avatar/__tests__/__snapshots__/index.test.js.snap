// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> should render a <Avatar/> components 1`] = `
<div>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-circle cuke-avatar-size-default cuke-avatar-icon"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
      />
      <circle
        cx="12"
        cy="7"
        r="4"
      />
    </svg>
  </span>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-circle cuke-avatar-size-default"
    text="黄瓜ui"
  >
    <span
      class="cuke-avatar-text"
    >
      黄瓜ui
    </span>
  </span>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-circle cuke-avatar-size-default cuke-avatar-image"
  >
    <img
      alt="cuke-avatar"
      src="https://www.test.jpg"
    />
  </span>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-circle cuke-avatar-size-default cuke-avatar-icon"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
      />
      <circle
        cx="12"
        cy="7"
        r="4"
      />
    </svg>
  </span>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-square cuke-avatar-size-default cuke-avatar-icon"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
      />
      <circle
        cx="12"
        cy="7"
        r="4"
      />
    </svg>
  </span>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-circle cuke-avatar-size-large"
    text="大"
  >
    <span
      class="cuke-avatar-text"
    >
      大
    </span>
  </span>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-circle cuke-avatar-size-default"
    text="中"
  >
    <span
      class="cuke-avatar-text"
    >
      中
    </span>
  </span>
  <span
    alt="cuke-avatar"
    class="cuke-avatar cuke-avatar-circle cuke-avatar-size-small"
    text="小"
  >
    <span
      class="cuke-avatar-text"
    >
      小
    </span>
  </span>
</div>
`;
